bitcoin/src/Bitcoin/Transaction/Builder.hs

```diff
@@ -273,9 +273,10 @@ verifyStdInput net tx i so0 val
     | isSegwit so0 =
         fromRight False $ (inp == mempty &&) . verifySegwitInput so0 <$> wp so0
     | otherwise =
-        fromRight False $
-            (verifyLegacyInput so0 <$> decodeInputBS net inp)
-                <|> (nestedScriptOutput >>= \so -> verifyNestedInput so0 so <$> wp so)
+        either
+            (const (fromRight False (nestedScriptOutput >>= \so -> verifyNestedInput so0 so <$> wp so)))
+            (verifyLegacyInput so0)
+            (decodeInputBS net inp)
```

Comment:
It's fine this time around, but I'd love if this was broken up into a separate commit in the future for audit/review. The review header you gave was definitely sufficient though to inform me what this was for so no need this time.